### DISCUSSION TITLE
snippeting with multiple matches

### DISCUSSION THREAD

**Discussion by bnfd** [algolia/instantsearch#4762] [Q&A]
I'm using `attributesToSnippet: ["content:60"]` to return a portion of the matched attribute. In cases where there are multiple matches (multiple occurrences of the keyword) in the same record, only the first one will be returned.
Is there a way to return all of them?

**Comment by francoischalifour** [comment]
The Algolia engine is responsible for returning the best match (that has the highest score), so it cannot return multiple snippeted parts.

I guess you could still compute and display them in the frontend but it would defeat the purpose of the Algolia backend search engine.

Can you tell us more why you would like to display multiple snippeted parts?

**Reply by bnfd** [reply]
By "compute" do you mean return the whole record and implement "snippets" in the frontend? 
Sometimes in a long document, there is the same keyword appearing multiple times and showing one snippet is not enough to identify the document. By providing multiple snippets (all the occurrences where the keyword exists in the document) we can get context.

**Reply by francoischalifour** [reply]
Get it. Another solution would be to split the document content in multiple attributes and to snippet all these attributes.
